src/workerd/api/pyodide/pyodide.c++

```diff
@@ -300,13 +300,76 @@ kj::Array<kj::String> ArtifactBundler::parsePythonScriptImports(kj::Array<kj::St
     }
   }
 
-  // XXX: jsg doesn't support kj::Vector return types, so this seems to be the only way to do this.
-  auto builder = kj::heapArrayBuilder<kj::String>(result.size());
-  for (auto i = 0; i < result.size(); i++) {
-    builder.add(kj::mv(result[i]));
+  return result.releaseAsArray();
+}
+
+// This is equivalent to `pkgImport.replace('.', '/') + ".py"`.
+kj::String importToModuleFilename(kj::StringPtr pkgImport) {
+  auto result = kj::heapString(pkgImport.size() + 3);
+  for (auto i = 0; i < pkgImport.size(); i++) {
+    if (pkgImport[i] == '.') {
+      result[i] = '/';
+    } else {
+      result[i] = pkgImport[i];
+    }
   }
 
-  return builder.finish();
+  result[pkgImport.size()] = '.';
+  result[pkgImport.size() + 1] = 'p';
+  result[pkgImport.size()] = 'y';
```

Comment:
This seems off, shouldn't it be 
```suggestion
  result[pkgImport.size() + 2] = 'y';
```

Reply:
Good catch! Indicates a bug in our validator-upload test which I've also fixed.